CSharpSourceCode/Abilities/Ability.cs

```diff
@@ -118,15 +118,15 @@ public virtual void ActivateAbility(Agent casterAgent)
 
             var frame = GetSpawnFrame(casterAgent);
 
-            var entity = SpawnEntity();
-            entity.SetGlobalFrame(frame);
+            GameEntity parentEntity = GameEntity.CreateEmpty(Mission.Current.Scene, false);
+            parentEntity.SetGlobalFrame(frame);
 
-            AddLight(ref entity);
+            AddLight(ref parentEntity);
 
             if (IsDynamicAbility())
-                AddPhysics(ref entity);
+                AddPhysics(ref parentEntity);
```

Comment:
Since parentEntity is an empty entity, it will never have a physicsbody, therefore AddPhysics will either do nothing or crash. I am still not a fan of doing the parentEntity, childEntity thing.

Reply:
what do we need physics for here?

Reply:
physics works properly

Reply:
awesome